Automattic/woocommerce-payments
# Fix PHPCS report in checkout classes

Fixes https://github.com/Automattic/woocommerce-payments/issues/8466

> Note: The command to run which will check files listed in https://github.com/Automattic/woocommerce-payments/issues/8466
```text
./vendor/bin/phpcs --standard=phpcs.xml.dist --sniffs='WordPress.Security.EscapeOutput,WordPress.Security.ValidatedSanitizedInput' includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php includes/class-wc-payments-checkout.php includes/class-wc-payments-payment-request-button-handler.php
```

#### Changes proposed in this Pull Request
This PR
* adds `isset` checks when getting the request param in `class-wc-payments-express-checkout-button-display-handler.php`
* mutes the phpcs report for non-escaped output in `class-wc-payments-checkout.php` when using `WC_Payments_Utils::esc_interpolated_html()` which itself escapes the return value 
* escapes the error message in `class-wc-payments-payment-request-button-handler.php`
* explicitly casts the product ID injected into the error message string



#### Testing instructions
Since the changes with`phpcs` reports were reverted, we need to work this around a little bit.

* `git fetch origin bc8ae14f29921f43f105a2240d21fe5c23ee932a:bc8ae14f29921f43f105a2240d21fe5c23ee932a`
* `git checkout bc8ae14f29921f43f105a2240d21fe5c23ee932a`
* run the command provided above and confirm that you see the errors for every file (5 errors overall)
* `git merge fix/phpcs-reports-checkout`
* run the command once more and no errors should appear

-------------------


**Post merge**
